Marlin/src/lcd/extui/mks_ui/mks_hardware_test.cpp

```diff
@@ -36,51 +35,51 @@
 #include "../../../module/temperature.h"
 #include "../../../sd/cardreader.h"
 
-uint8_t pw_det_sta, pw_off_sta, mt_det_sta, mt_det3_sta;
-#if PIN_EXISTS(MT_DET_2)
-  uint8_t mt_det2_sta;
-#endif
-uint8_t endstopx1_sta, endstopx2_sta, endstopy1_sta, endstopy2_sta, endstopz1_sta, endstopz2_sta;
-void test_gpio_readlevel_L() {
-  #if ENABLED(MKS_TEST)
+#if ENABLED(MKS_TEST)
+
+  #include "mks_hardware_test.h"
+
+  bool pw_det_sta, pw_off_sta, mt_det_sta;
+  #if PIN_EXISTS(MT_DET_2)
+    bool mt_det2_sta;
+  #endif
+  bool endstopx1_sta, endstopx2_sta, endstopy1_sta, endstopy2_sta, endstopz1_sta, endstopz2_sta;
+
+  void test_gpio_readlevel_L() {
     volatile uint32_t itest;
     WRITE(WIFI_IO0_PIN, HIGH);
     itest = 10000;
-    while (itest--);
-    pw_det_sta = (READ(MKS_TEST_POWER_LOSS_PIN) == 0);
-    pw_off_sta = (READ(MKS_TEST_PS_ON_PIN) == 0);
-    mt_det_sta = (READ(MT_DET_1_PIN) == 0);
+    while (itest--); // Counting to 10000 takes some amount of time
```

Comment:
It may be better to use a low level delay function here, since we don't know the speed of the MCU.